flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterDescriptor.java

```diff
@@ -1568,151 +1568,78 @@ public static ApplicationSubmissionContextReflector getInstance() {
-            Method applicationTagsMethod;
-            Method attemptFailuresValidityIntervalMethod;
-            Method keepContainersMethod;
-            Method nodeLabelExpressionMethod;
-
-            try {
-                // this method is only supported by Hadoop 2.4.0 onwards
-                applicationTagsMethod = clazz.getMethod(APPLICATION_TAGS_METHOD_NAME, Set.class);
-                LOG.debug(
-                        "{} supports method {}.",
-                        clazz.getCanonicalName(),
-                        APPLICATION_TAGS_METHOD_NAME);
-            } catch (NoSuchMethodException e) {
-                LOG.debug(
-                        "{} does not support method {}.",
-                        clazz.getCanonicalName(),
-                        APPLICATION_TAGS_METHOD_NAME);
-                // assign null because the Hadoop version apparently does not support this call.
-                applicationTagsMethod = null;
-            }
-
-            this.applicationTagsMethod = applicationTagsMethod;
-
-            try {
-                // this method is only supported by Hadoop 2.6.0 onwards
-                attemptFailuresValidityIntervalMethod =
-                        clazz.getMethod(ATTEMPT_FAILURES_METHOD_NAME, long.class);
-                LOG.debug(
-                        "{} supports method {}.",
-                        clazz.getCanonicalName(),
-                        ATTEMPT_FAILURES_METHOD_NAME);
-            } catch (NoSuchMethodException e) {
-                LOG.debug(
-                        "{} does not support method {}.",
-                        clazz.getCanonicalName(),
-                        ATTEMPT_FAILURES_METHOD_NAME);
-                // assign null because the Hadoop version apparently does not support this call.
-                attemptFailuresValidityIntervalMethod = null;
-            }
-
-            this.attemptFailuresValidityIntervalMethod = attemptFailuresValidityIntervalMethod;
-
-            try {
-                // this method is only supported by Hadoop 2.4.0 onwards
-                keepContainersMethod = clazz.getMethod(KEEP_CONTAINERS_METHOD_NAME, boolean.class);
-                LOG.debug(
-                        "{} supports method {}.",
-                        clazz.getCanonicalName(),
-                        KEEP_CONTAINERS_METHOD_NAME);
-            } catch (NoSuchMethodException e) {
-                LOG.debug(
-                        "{} does not support method {}.",
-                        clazz.getCanonicalName(),
-                        KEEP_CONTAINERS_METHOD_NAME);
-                // assign null because the Hadoop version apparently does not support this call.
-                keepContainersMethod = null;
-            }
-
-            this.keepContainersMethod = keepContainersMethod;
+            this.applicationTagsMethod = getMethod(clazz, APPLICATION_TAGS_METHOD_NAME, Set.class);
```

Comment:
Please add the previously existing comments：
// this method is only supported by Hadoop 2.4.0 onwards